src/audio/eq_iir.c

```diff
@@ -143,99 +130,128 @@ static void eq_iir_free_delaylines(struct iir_state_df2t *iir)
 	 * allocated buffer
 	 */
 	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
-		if ((iir[i].delay != NULL) && (delay == NULL))
+		if (iir[i].delay && !delay)
 			delay = iir[i].delay;
 
 		/* Point all delays to NULL to avoid duplicated free later */
 		iir[i].delay = NULL;
 	}
 
-	if (delay != NULL)
+	if (delay)
```

Comment:
in Linux we can free without testing, can't remember if this is required for SOF?

Reply:
Good, free_block() that rfree() uses does this check so this check is not needed.